### PR TITLE
#3802 - replace null parameter with containingType

### LINKED ISSUE #3802: Bug: ContainingType parameter isn't passed along instead null is used
https://github.com/MarimerLLC/csla/blob/881c262f695cfa02b7e936f123280aadd5fb867f/Source/Csla/PropertyInfo.cs#L104-L105

This looks like a bug to me. The third parameter `null` in `this()` is actually `Type containingType`. But why isn't the parameter passed along but instead `null` used?
Is this a bug or intended?

### PR DESCRIPTION
**Pull request by 19bibo85**
Fixes #3802  - Replace null parameter with containingType.

### CODE REVIEW

**Comment by SimonCropp** [comment]
is it possible to add a test for this?

**Comment by StefanOssendorf** [comment]
And did you sign the contribution agreement and send it to Rocky?

**Comment by rockfordlhotka** [comment]
> And did you sign the contribution agreement and send it to Rocky?

Yes, I have the agreement.

**Comment by 19bibo85** [comment]
@SimonCropp Test `TestContainingType` has been added to the `PropertyInfoTests` class.

File: Source/Csla.test/PropertyInfo/PropertyInfoTests.cs [hunk 1]
```diff
@@ -69,5 +69,12 @@ public void TestStringNullDefaultValue()
       Assert.AreEqual(null, PropertyInfoRoot.StringNullDefaultValueProperty.DefaultValue);
       Assert.AreEqual(null, PropertyInfoRoot.NewPropertyInfoRoot(dataPortal).StringNullDefaultValue);
     }
+
+    [TestMethod]
+    public void TestContainingType()
```
Comment (StefanOssendorf) [inline Source/Csla.test/PropertyInfo/PropertyInfoTests.cs:74]: As already written as a comment n the PR. This test does not check the fix from `PropertyInfo.cs`.
When you revert the fix the test is still green.

**Review by StefanOssendorf — CHANGES_REQUESTED** [review]

File: Source/Csla.test/PropertyInfo/PropertyInfoTests.cs (hunk 1 above)
Comment (19bibo85) [reply Source/Csla.test/PropertyInfo/PropertyInfoTests.cs:74]: @StefanOssendorf  Thank you for reviewing the PR. You're right, I have just amended the `ContainingTypeProperty` using the `PropertyInfo` constructor changed. I confirm that reverting the change the test now fails. :-)

**Review by StefanOssendorf — APPROVED** [review]

**Pull request merged**